### ISSUE TITLE
Different name (path) while extract same sound file and subtitles.

### ISSUE THREAD

**Issue by kaikaiiiiiii** [overtools/OWLib#54]
For example, to extract subtitles with command

`DataTool.exe -L=enUS -T=enUS "..\Overwatch" list-subtitles >enlines.txt`

you can see

**00000000BE3F.071
    The will-o'-the-wisp flashed through the tide of Dr. Junkenstein's minions, moving from one to the next faster than the eye could see.**

to extract sound file with command 

`tools\DataTool.exe -L=enUS -T=enUS "..\Overwatch" extract-maps .\extracted\ *`

you can find

**"Maps\Junkenstein's Revenge\6C9\VoiceSets\00000000013B.05F\00000000058E.078\000000054F09.0B2.ogg"**

The two should be same, the sound file plays the line which subtitle wrote. And there's only one result that matches each other.

But as the above shows, there's no clues to match these two.

I played the OGG file and find these by hand.

**Comment by Js41637** [comment]
Can't really understand what you're saying but there is 2 list subtitles commands. `list-subtitles` only lists the subtitles and the id of the string (`00000000BE3F.071`). `list-subtitles-real` lists all voicelines that have a subtitle attached to it which is what you are looking for. These commands are listed in the help when you run DataTool.

```
  list-subtitles          | List subtitles
  list-subtitles-real     | List subtitles (from audio data)
```

**Comment by kaikaiiiiiii** [comment]
Edit: 

I see. So my question changes to:

Is there any relationship between the $id of the subtitle string and the $filename or $foldername of the ogg file?

--------------------------------------------

I don't compare two commands, I mean the extracted subtitle can't match to ogg file.

Let me try again. I put an example what I want:

**"HeroVoice\McCree\Default\000000000051.05F\000000000969.078\00000004A547.0B2.ogg"** 

I got this file by extract-hero-voice. It's a sound file, and if you play it, it says "And getting shot at"

Then I extract subtitles-real and find this line of text.

**00000000019F.05F: 00000004A57E.0B2 - And getting shot at.**

They are same, exactly the same, I can pair to each other by the same string of **00000004A57E.0B2**

This is what I mean **can match**.

There are many ogg files in retribution event can match to subtitle texts, which makes game video creators much easy to find sounds they need, to put into their videos.

-----------------------------------------------

Then here is the failed example:

Maps\Junkenstein's Revenge\6C9\VoiceSets\00000000013B.05F\00000000058E.078\000000054F09.0B2.ogg

00000000BE3F.071
The will-o'-the-wisp flashed through the tide of Dr. Junkenstein's minions, moving from one to the next faster than the eye could see.

That ogg file, this section of subtitle text, they should be a pair, cause the subtitle shows exactly what the OGG file says. But they have not clues to show their relationship to each other.

To ogg file, I can only get  00000000013B.05F, 00000000058E.078, 000000054F09.0B2 three strings,

to the subtitle I can only get 00000000BE3F.071. 

These strings doesn't match, which means I can't find out which subtitle line can pair to an ogg file.

**Comment by Js41637** [comment]
If you read what I said in my message above, I said;

> **NOTE: Not all subtitles in list-subtitles will be in list-subtitles-real. This is an issue with the game and not something we have any control over**

**Comment by kaikaiiiiiii** [comment]
> If you read what I said in my message above, I said;
> 
> > **NOTE: Not all subtitles in list-subtitles will be in list-subtitles-real. This is an issue with the game and not something we have any control over**
> 
> List Subtitles just lists all "subtitles" in the game, just because they are there doesn't mean they are used, most of the subtitles in list-subtitles are **not** used meaning the tools have no way of linking the subtitle to the voiceline. 071 is just a string, on it's own it has no context to anything else.

Thanks, I see. I'll try any speech recognize API to convert them.
